src/js/classes/Graph.js

```diff
@@ -216,10 +204,85 @@ export class ScatterPlot {
     this.renderAxis(this.xAxis, false);
     this.renderAxis(this.yAxis, true);
 
-    // order: lines in background, then points, then names on top
-    this.points.forEach(point => point.renderLine());
-    this.points.forEach(point => point.renderPoints());
-    this.points.forEach(point => point.renderCountyName());
+    // order: lines in background, then points sorted by radius, then names on top
+    this.renderLines();
+    this.renderPointsInOrder();
+    this.renderCountyNames();
+  }
+
+  renderLines() {
+    this.points.forEach((point) => {
+      // only draw line if we have two data points
+      if (point.data.length != 2) return;
+
+      const className = `scatter-line${point.outlier ? " outlier" : ""}`;
+      const line = document.createElementNS(SVG_NS, "line");
+      line.setAttributeNS(null, "class", className);
+      line.setAttributeNS(null, "x1", point.xs[0]);
+      line.setAttributeNS(null, "y1", point.ys[0]);
+      line.setAttributeNS(null, "x2", point.xs[1]);
+      line.setAttributeNS(null, "y2", point.ys[1]);
+      this.plot.appendChild(line);
+      point.elements.push(line);
+
+      const hoverLine = document.createElementNS(SVG_NS, "line");
+      hoverLine.setAttributeNS(null, "class", `${className} hover-line`);
+      hoverLine.setAttributeNS(null, "x1", point.xs[0]);
+      hoverLine.setAttributeNS(null, "y1", point.ys[0]);
+      hoverLine.setAttributeNS(null, "x2", point.xs[1]);
+      hoverLine.setAttributeNS(null, "y2", point.ys[1]);
+      hoverLine.setAttributeNS(null, "y2", point.ys[1]);
+      this.plot.appendChild(hoverLine);
+
+      hoverLine.addEventListener("mouseenter", () => point.onMouseEnter());
+      hoverLine.addEventListener("mouseleave", () => point.onMouseLeave());
+    });
+  }
+
+  renderCountyNames() {
```

Comment:
See my comment above; all of this should be moved back into the `CountyPoint` class.